### PR TITLE
[Enhancement] Allow nickname assignment in the starter select UI.

### PR DESCRIPTION
**Pull request by KimJeongSun**
## What are the changes?

In the starter select UI, it is now possible to set nicknames for Pokémon!

## Why am I doing these changes?

Many users have requested this feature (myself included!).

## What did change?

1. Added a rename menu for setting nicknames.
2. Included a nickname attribute in the starter settings, which will be used during the `tryStart` process.
3. Assigned this feature to integrate with the current nickname functionality.
( so, The rename settings in the start UI are only used for initializing the starter party in a new run. )

### Screenshots/Videos

https://github.com/user-attachments/assets/0c6ebf96-b9a2-4bda-844e-cff7d8dc5c44

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/ui/rename-form-ui-handler.ts [hunk 1]
```diff
@@ -36,17 +36,13 @@ export default class RenameFormUiHandler extends FormModalUiHandler {
   show(args: any[]): boolean {
     if (super.show(args)) {
       const config = args[0] as ModalConfig;
-      this.inputs[0].text = (args[1] as PlayerPokemon).getNameToRender();
-
+      if (args[1] && typeof (args[1] as any).getNameToRender === "function") {
```
Comment (torranx) [inline src/ui/rename-form-ui-handler.ts:39]: why is this casted as `any`? shouldn't it be `PlayerPokemon` too?

File: src/ui/starter-select-ui-handler.ts [hunk 2]
```diff
@@ -2573,7 +2601,13 @@ export default class StarterSelectUiHandler extends MessageUiHandler {
 
     if (species && (this.speciesStarterDexEntry?.seenAttr || this.speciesStarterDexEntry?.caughtAttr)) {
       this.pokemonNumberText.setText(Utils.padInt(species.speciesId, 4));
-      this.pokemonNameText.setText(species.name);
+      // if starterAttributes nickname is set, use it, otherwise use species name
```
Comment (torranx) [inline src/ui/starter-select-ui-handler.ts:2604]: this can be removed imo, the code is pretty straightforward and readable

Comment (KimJeongSun) [reply src/ui/starter-select-ui-handler.ts:2604]: OK

**Comment by damocleas** [comment]
Does this stay at the start of a new session? If I nickname a Pikachu "Bolt" and I leave the menu does it stick?

**Comment by KimJeongSun** [comment]
> Does this stay at the start of a new session? If I nickname a Pikachu "Bolt" and I leave the menu does it stick?

Sure, it gets saved along with the other starter settings. Even if you refresh or return to the main menu, the nickname will be retained!

**Pull request merged**